server/src/service.rs

```diff
@@ -256,9 +258,19 @@ impl Service {
         let object_server = connection.object_server();
         let service = Self {
             collections: Default::default(),
+            sessions: Default::default(),
             manager: Arc::new(Mutex::new(ServiceManager::new(connection.clone()))),
         };
 
+        {
+            let service_clone = Arc::new(service.clone());
```

Comment:
What is the use of Arc here id the service is clonable anyways?

Reply:
Mmm.. we can't `Arc::clone` without a `Arc::new` right?

Reply:
Why you need Arc::clone anyways?

Reply:
Cloning the service would just increment the ref count of every instance in it, so there is no need for another Arc wrapper.

Reply:
Oh, I see. Thanks!

Reply:
A quick question: Isn't Arc::clone is really fast and uses less memory compared to a normal clone? And putting a Arc wrapped Service in ServiceManager might be good, because we're accessing it many times in different places. May be I'm wrong :)

Reply:
This is not a normal clone though. Clone can only be implemented if all the inner items implement the Clone traits. In this case, all the inner items are wrapped in Arc. So basically you will get the same thing without the extra Arc.

Reply:
Yes. Understood, thanks :)

Reply:
Thanks again for the explanation. Done.